server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1619,6 +1634,7 @@ public void canMatch(CanMatchNodeRequest request, ActionListener<CanMatchNodeRes
         final List<CanMatchNodeResponse.ResponseOrFailure> responses = new ArrayList<>(shardLevelRequests.size());
         for (var shardLevelRequest : shardLevelRequests) {
             try {
+                // TODO remove the exception handling as it's now in canMatch itself
                 responses.add(new CanMatchNodeResponse.ResponseOrFailure(canMatch(request.createShardSearchRequest(shardLevelRequest))));
```

Comment:
I left this for a follow-up as I need to add a new transport version and bw comp handling.